src/libs/HttpUtils.js

```diff
@@ -98,7 +99,16 @@ function xhr(command, data, type = CONST.NETWORK.METHOD.POST, shouldUseSecure =
 
     let apiRoot = shouldUseSecure ? CONFIG.EXPENSIFY.SECURE_EXPENSIFY_URL : CONFIG.EXPENSIFY.URL_API_ROOT;
 
-    if (CONFIG.IS_IN_STAGING && shouldUseStagingServer) {
+    // getEnvironment promise returns the environment based on ENVIRONMENT Config for web/desktop.
+    // Additionally for ios/android betaChecker.isBetaBuild() does the version check
+    // environment is initialized as null as the default set in CONFIG.js always takes precedence
+    let environment = null;
+    Environment.getEnvironment()
+        .then((env) => {
+            environment = env;
+        });
+
+    if (environment !== CONST.ENVIRONMENT.PRODUCTION && shouldUseStagingServer) {
```

Comment:
@mountiny I introduced the usage of staging api_url in DEV environment. Earlier it was only for staging. Is it ok ?

Reply:
Not sure what you mean here., can you try to rephrase you question please?